beluga_ros/include/beluga_ros/ndt_ellipsoid.hpp

```diff
@@ -0,0 +1,132 @@
+// Copyright 2024 Ekumen, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BELUGA_ROS_NDT_ELLIPSOID_HPP
+#define BELUGA_ROS_NDT_ELLIPSOID_HPP
+
+#include <beluga_ros/messages.hpp>
+
+/**
+ * \file
+ * \brief Utilities for NDT I/O over ROS interfaces.
+ */
+
+namespace beluga_ros {
+
+static void use_mean_covariance(const auto& mean, const auto& eigenSolver, auto& marker){
+  marker.type = beluga_ros::msg::Marker::SPHERE;
+  marker.action = beluga_ros::msg::Marker::ADD;
+  
+  // Compute the rotation based on the covariance matrix
+  const auto eigenvectors = eigenSolver.eigenvectors().real();
+  const auto eigenvalues = eigenSolver.eigenvalues();
+
+  Eigen::Matrix3d rotationMatrix;
+  Eigen::Vector3d scalevector;
+  rotationMatrix << eigenvectors.col(0), eigenvectors.col(1), eigenvectors.col(2);
+  scalevector = Eigen::Vector3d{eigenvalues.x(), eigenvalues.y(), eigenvalues.z()};
+
+  // Permutation
+  if (std::abs(rotationMatrix.determinant() - 1.0) > Eigen::NumTraits<double>::dummy_precision()) {
+    rotationMatrix << eigenvectors.col(1), eigenvectors.col(0), eigenvectors.col(2);
+    scalevector = Eigen::Vector3d{eigenvalues.y(), eigenvalues.x(), eigenvalues.z()};
+  }
```

Comment:
@Fernando-Sanz why do we need to change the determinant sign by permuting the matrix?

Reply:
Because we want only a rotation as transformation. I noticed that this way it produces more reasonable results (e.g. some ellipsoids get aligned with other in walls or the floor). Maybe that was a wrong behaviour, is it a bad idea?
![ndt_map_visualization_no_permutation](https://github.com/user-attachments/assets/485bede9-6815-4711-9428-cbcf6056f395)
> Obstacle representation with no permutation. Note that some of the ellipsoids in the floor and walls are not aligned in contrast with the same ellipsoids in the main description picture.